starters/next-react-query-tailwind/src/components/Greeting/Greeting.data.tsx

```diff
@@ -3,34 +3,17 @@ import { GreetingView } from './Greeting.view';
 
 export function Greeting() {
   const {
-    data: greeting,
+    data: message,
     isLoading,
-    refetch,
     error,
-  } = useQuery<{ message: string }>('hello', async () => {
-    return await fetch('http://localhost:3000/api/hello').then((res) =>
-      res.json()
-    );
+  } = useQuery<string>('hello', async () => {
+    const response = await fetch('https://api.starter.dev/hello?greeting=from This Dot Labs!');
+    return await response.text();
   });
 
-  const handleGreetingClick = () => {
-    refetch();
-  };
-
-  if (isLoading) {
-    return (
-      <div
-        data-testid="loading"
-        className="absolute top-0 left-0 w-full h-full bg-white bg-opacity-25 animate-pulse"
-      />
-    );
-  }
-
-  if (error || !greeting) {
-    return null;
-  }
+  const displayMessage = error ? null : message;
```

Comment:
This is a bit of a react smell. It would be better to pass the error into the view to be handled as a display OR if we really just don't want to display anything when there's an error, you could just do `if (error) { return null; }` here. I'd opt for showing the error message in the view component.